gen

```diff
@@ -124,10 +124,11 @@ check_ext() {
 
 exists() {
     if [[ -e "$FILENAME" ]]; then
-        printf "Filename already exists.\n"; exit
+        printf "Filename already exists.\n"; exit 1
     else
         if ! touch "$FILENAME" 2> /dev/null; then
-            usage
+            printf "Error: Couldn't touch \"$FILENAME\"\n\n" 
```

Comment:
I don't want this to print that it can't touch that's why I have error 2> /dev/null. If I removed that it would do what your printf is doing. The only time it should fail to touch is if the user inputs something incorrectly, in which case I want "usage" to be printed.

Reply:
> The only time it should fail to touch is if the user inputs something incorrectly

That is one case, but it also may be that the user doesn't have permission to touch a file (in a specific directory, or even at all)?

Additional possible issues may relate to symlinks in the path (edge cases involving cyclic graphs, etc), filled up hardware, filesystem errors, and so on

Shouldn't we keep it general where we can? I'd say it might be better to be honest about errors and still print usage if necessary, because the errors definitely don't just relate to user input.

Reply:
Fair enough, a better solution would be to remove the redirection since touch will already echo an error. Thanks.

Reply:
Coolio, [ffcdb4](https://github.com/Ekatwikz/gen/commit/ffcdb4f) does this now then, I think github will add it to this pull request